docs/DeveloperGuide.md

```diff
@@ -1,34 +1,330 @@
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[3.1 Target User Persona](#userpersona)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[3.2 Target User Profile](#userprofile)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[3.3 Value Proposition](#value)
+#### [4. User Stories](#userstories)
+#### [5. Non-Functional Requirements](#nfr)
+#### [6. Glossary](#gloss)
+#### [7. Instructions for Manual Testing](#testinstr)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<br>
 
+## <a id="intro">1. Introduction</a>
 
-## Product scope
-### Target user profile
+NotUS is a quick and simple, Command Line Interface (CLI) based, note-taking application for keyboard-inclined users. NotUS allows for users to categorize notes by tagging as well as pinning the more important notes. NotUS is also designed to assist in planning timetables to highlight possible clashes.
 
-{Describe the target user profile}
+This document describes the design, implementation and architecture of NotUS. The aim of this developer guide is to get developers and potential contributors familiarised with the design and implementation of NotUS. It is assumed that the reader has some basic understanding of UML Notations. If you do not possess such knowledge, this document is probably not meant for you. Please access the [User Guide](https://github.com/AY2021S1-CS2113-T13-1/tp/blob/master/docs/UserGuide.md) instead.
 
-### Value proposition
+#### <a id="setup"><ins>1.1 Setting Up</ins></a>
 
-{Describe the value proposition: what problem does it solve?}
+**Prerequisites:**
+* JDK 11
+* IntelliJ IDE
 
-## User Stories
+Fork this repo and clone it onto your local machine.
+Import the project as a **Gradle project**.
+Ensures that you are using the correct JDK version (For this project we are using JDK 11).
 
-|Version| As a ... | I want to ... | So that I can ...|
+For a more detailed set of instructions, please refer to the [following guide](https://github.com/AY2021S1-CS2113-T13-1/tp/blob/master/README.md).
+
+<br>
+
+## <a id="design">2. Design & Implementation</a>
+
+This section seeks to explain the high-level design of the application. Given below is a quick overview of each component and the explanation of the design architecture in greater detail. NotUS is the main class of the application, and handles the initializing and execution of the appropriate classes. <br>
+Diagrams found in our documentation were generated using <a href="https://plantuml.com/">PlantUML</a> and references were made to <a href="https://github.com/se-edu/addressbook-level2/tree/master/src/seedu/addressbook">addressbook-level2</a> for the structure of the classes and packages. The structures have been modified to meet the needs of our application.
+
+#### <a id="overview"><ins>2.1 Architecture Overview</ins></a>
+
+<p align="center">
+  <img alt="NotUS Architecture Overview" src="diagrams/out/Architecture_Overview.png" />
+  <br><em>Figure 1</em>
+</p>
+
```

Comment:
Since it is an architecture diagram, you may consider to combine the `ParserManager` and `Command` as Logic component.